src/data/polynomial/iterated_derivative.lean

```diff
@@ -0,0 +1,163 @@
+/-
+Copyright (c) 2020 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import data.polynomial.derivative
+import logic.function.iterate
+import tactic.ring tactic.linarith
+
+/-!
+# Theory of iterated derivative
+We define and prove some lemmas about iterated (formal) derivative for polynomials over a semiring.
+-/
+
+noncomputable theory
+
+open finset nat polynomial
+open_locale big_operators
+
+namespace polynomial
+universes u
+variable {R : Type u}
+
+section semiring
+
+variables [semiring R] (f p q: polynomial R) (n k : ℕ)
+
+/--Defining the n-th derivative of polynomial f-/
+def iterated_derivative : polynomial R := derivative ^[n] f
```

Comment:
As you can see below, the downside of a long name for a definition is that statements that use it 3 times become pretty long. Hence I would prefer `iterated_deriv`. (That can save 15 chars on a line that uses it 3 times.)

Reply:
Should we give this a notation?